src/lseq/ident.rs

```diff
@@ -134,8 +122,8 @@ impl<A: Actor> IdentGen<A> {
                 // The two paths are fully equal which means that the site_ids MUST be different or
                 // we are in an invalid situation
                 (None, None) => {
-                    let max_for_depth = self.width_at(depth as u32) - 1;
```

Comment:
I think this `- 1` here was a bug once again.

`depth` has been incremented to the point that we have exceeded both p and q. hence `depth` is currently at the correct depth that we want to allocate for.